refinery/tool_manager/models.py

```diff
@@ -449,40 +456,127 @@ class Meta:
+        Handles the association of Galaxy objects with their parent elements
+        :param galaxy_element_data: list of bioblend objects to associate
+        with each other
+        :return: <CollectionDescription>
+        """
+        if not len(galaxy_element_data) == 1:
+            for index, galaxy_element in enumerate(galaxy_element_data):
+                if type(galaxy_element) == CollectionDescription:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
+                        if (type(prev_galaxy_element) ==
+                                HistoryDatasetElement or
+                                type(prev_galaxy_element) ==
+                                CollectionElement):
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+                if type(galaxy_element) == CollectionElement:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
+                        if type(prev_galaxy_element) == HistoryDatasetElement:
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+                        if (type(prev_galaxy_element) == CollectionElement and
+                                prev_galaxy_element.type !=
+                                galaxy_element.type):
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+        else:
+            assert len(galaxy_element_data) == 1
+            assert type(galaxy_element_data[0]) == CollectionDescription
+            return galaxy_element_data[0]
+
+    def _create_collection_description(self, galaxy_element_list=None):
+        """
+        Creates an appropriate bioblend.galaxy.CollectionDescription
+        instance based off of the structure of our WorkflowTool's
+        file_relationships in a recursive manner.
+        :return: <CollectionDescription>
+        """
+        file_relationship_nesting_list = (
+            self._parse_file_relationships_nesting(
+                *self.get_file_relationships_galaxy()
+            )
+        )
+
+        if galaxy_element_list is None:
+            galaxy_element_list = []
+
+        for nested_element in reversed(file_relationship_nesting_list):
+            if isinstance(nested_element, dict):
+                galaxy_element_list.append(
```

Comment:
There should be logic here to account for `forward` and `reverse` `HistoryDatasetElement`s when the `CollectionType` is `paired`